query/src/sql/planner/binder/scalar_common.rs

```diff
@@ -115,3 +118,12 @@ pub fn split_equivalent_predicate(scalar: &Scalar) -> Option<(Scalar, Scalar)> {
         _ => None,
     }
 }
+
+pub fn wrap_cast(scalar: Scalar, target_type: &DataTypeImpl) -> Scalar {
+    let cast = CastExpr {
```

Comment:
Add if `scalar.data_type() != target_type` special case

Reply:
Is `DataType` comparable? 

At the beginning, I was trying to use `DataTypeID` to compare the two types, but found it doesn't work with nested data types.

Maybe I should compare them with type names?

Reply:
Yes, it's comparable.

```
impl PartialEq for DataTypeImpl {
    fn eq(&self, that: &DataTypeImpl) -> bool {
        equal(self, that)
    }
}
```

Reply:
@sundy-li Fixed.